mne/gui/__init__.py

```diff
@@ -247,6 +247,48 @@ def locate_ieeg(info, trans, aligned_ct, subject=None, subjects_dir=None,
     return gui
 
 
+@verbose
+def view_stc(stc, subject=None, subjects_dir=None, src=None,
+             inst=None, show=True, block=False, verbose=None):
+    """Locate intracranial electrode contacts.
+
+    Parameters
+    ----------
+    stc : ndarray of shape (n_epochs, n_vertices, n_orient, n_freqs, n_times)
+        The source estimate data.
+    %(subject)s
+    %(subjects_dir)s
+    src : instance of SourceSpaces
+        The volume source space for the ``stc``.
+    inst : EpochsTFR | AverageTFR | Epochs | Evoked | None
```

Comment:
I thought the scope for this GSoC was specifically for viewing TFRs in a brain (somehow!) but now it seems much more general if you can pass `Epochs` or `Evoked`. Based on the naming and inputs it seems a lot like either one or both of:

1. `stc.plot` / `Brain(...).add_data(...)`, which provides a 3D view.
2. `vol_stc.plot` / `plot_volume_source_estimate`, which gives orthoview slices of 3D volumes

This makes me worry a bit about scope / vision / clarity of uses for end-users (who, up until now, should just work with some variant of `Brain`). Have you already had discussions on GH about this overlap and how to deal with it that I should read?

It's always easier and more fun to make something from scratch, but there is a lot of value in integrating with existing stuff where possible, so I want to make sure we have where we can.

You also say you support Epochs and Evoked but they are not used anywhere in an example/tutorial. This would help me see the utility of supporting these.

To me the quick fix is to keep this TFR-specific and adjust the naming (and input args) accordingly

Reply:
That sounds fine. They can always be added later if needed. The thought was that in the proposal, topography was asked to be supported and I thought it would be nice to allow any topography. I'll just stick with the TFR objects for now.

Reply:
So another point--I thought it would be nice since the structure of the GUI is so similar to `stc.plot` to allow that to use this GUI in a future PR. IMO, it's a lot cleaner than having things in `mne.viz.Brain` and having all the events and everything run through the renderer. If that's not a direction you want to go in, I probably am not going to be making a lot of those PRs so we can just leave it be. I wrote the code in a way that I thought would make it easy to do that because I thought it was a good idea.

Reply:
I'm not sure the best solution here. I just really don't want to end up with redundant code and ways of doing things. It's a recipe for cruft and poor maintenance internally, and confusion for end users.

> I thought it would be nice since the structure of the GUI is so similar to stc.plot to allow that to use this GUI in a future PR

At one point in the last couple of/few years we changed from `stc.plot` routing through PySurfer to routing through our own internal `mne.viz.Brain` instance. It took many work-hours of effort (maybe 1 year FTE?) but brought a lot of benefits. This makes me worried about the possibility of redoing things yet again in terms of hours of coding and review required for what benefits it will bring.

The safest approach for now I think is to leave open the possibility internally of expanding in this direction, but not make API promises that we have to maintain and explain (Evoked, Epochs)

Reply:
That sounds reasonable. I already implemented most of what `stc.plot` does and it's a lot easier this way, there aren't weird hacks with resizing the render window like in `mne.viz.Brain`. IMO, it would be a good improvement not too much cost since most of the code is already there for this GUI. The time-frequency is the most general case so it's much easier to just include time. Not that it's no work or that I'm proposing doing it all, it just seems like `mne.viz.Brain` isn't really the framework for this kind of GUI stuff IMO. I'm not wedded to the idea, just giving my two cents.